addinstanceform.php

```diff
@@ -67,4 +79,39 @@ protected function configuration_definition() {
         $form->setDefault('compressor', cachestore_redis::COMPRESSOR_NONE);
         $form->setType('compressor', PARAM_INT);
     }
+
+    /**
+     * Validates the configuration form data
+     *
+     * @param array $data
+     * @param array $files
+     * @return array
+     */
+    public function configuration_validation($data, $files, $errors) {
```

Comment:
It is called by `cachestore_addinstance_form` (the parent class) when 'validation` is called. If validation fails (returns an array with count > 0) then the form->getdata() returns null/false and the form is displayed again, with the errors highlighted on the form.